### ISSUE TITLE
Possible Bug: Calling render twice in controller

### ISSUE THREAD

**Issue by RickCarlino** [voltrb/volt#216]
I was trying to find a DRY way to handle documents not being found:

```ruby
require 'pry'
module Main
  class SensorReadingsController < Volt::HttpController
    def show
      render json: sensor_reading
    end

  private

    def sensor_reading
      @reading ||= readings.find_one(_id: params._id) || not_found
    end

    def not_found
      render json: {error: "Cant find sensor_reading with _id: #{params._id || 'none'}"}
    end
  end
end
```
but when I hit the controller action:

```
GET localhost:3000/sensor_readings/1
```

It returned:

```
{"error":"Cant find sensor_reading with _id: 1"}["{\"error\":\"Cant find sensor_reading with _id: 1\"}"]
```

**Comment by jfahrer** [comment]
```render``` get's called twice in your example. ```not_found``` should just return the hash

Currently render just sets the reponse headers and appends the array Rack expects the body in. I don't know if there is a usecase where you would append to the body multiple times. So maybe we can just throw an exception in render get's called multiple times.

**Comment by jfahrer** [comment]
Ah sorry, I think what you meant is, that calling ```render``` should actually render the response directly and stop processing.

**Comment by RickCarlino** [comment]
Yep, basically, it seems like we would need:

1. Some kind of `rescue_from` functionality (to keep the controller DRY).
2. A `find_one!` bang method that raises an exception if not found.
3. The ability to halt the rendering chain early.

Curious what @ryanstout has to say about what the best approach would be for this situation and if the solution exists or needs a PR.

**Comment by jfahrer** [comment]
I've got the before/after actions implemented. So you could:

```ruby
module Main
  class SensorReadingsController < Volt::HttpController
    before_action :set_sensor_reading, only: [:show]
    before_action :require_sensor_reading, only: [:show]

    def show
      render json: @sensor_reading
    end

  private

    def set_sensor_reading
      @sensor_reading ||= readings.find_one(_id: params._id)
    end

    def require_sensor_reading
      return if @sensor_reading
      render json: {error: "Cant find sensor_reading with _id: #{params._id || 'none'}"}
      stop_chain
    end
  end
end
```

There is some enhancements I'm working on for the http_controllers. Goiing to take a few more days..

**Comment by RickCarlino** [comment]
Ah! `stop_chain`. That's will work fine for what I'm trying to do. Thanks for the heads up.

**Comment by RickCarlino** [comment]
Looks like `stop_chain` and `before_action` are actually not available in 0.9.2. Was this added in a later version?

**Comment by ryanstout** [comment]
@RickCarlino any chance you can see if this is still happening?  I'm pretty sure I saw the same issue and fixed it.  Thanks!  Please reopen if it is broke still.
